main.bal

```diff
@@ -61,6 +61,29 @@ function getPort() returns int {
 # The final configured server port.
 final int & readonly serverPort = getPort().cloneReadOnly();
 
+# User specified host ip address of backend server (with protocol and port)
+# Can also be specified by setting the `QHANA_HOST` environment variable.
+configurable string host = "http://localhost:" + serverPort.toString();
+
+# Determine the base host url from the `QHANA_HOST` environment variable.
+# If not present use the configurable variable `host` as fallback.
+#
+# + return - the configured host base path (including protocol and port)
+function getHost() returns string {
+    string h = os:getEnv("QHANA_HOST");
+    if (regex:matches(h, "(?<=^|https?://)localhost(:[0-9]+)?")) {
```

Comment:
This regex is way too restrictive! Just ensure, that it starts with `http(s)://`